Conversion/MultiCopy.FCMacro

```diff
@@ -0,0 +1,1439 @@
+
+class Filter(QtCore.QObject):
+    """
+    This class has only one function which is responsible for validating 
+    the input paste code commands when the user focusses out of the input 
+    text box. Appropriately, the GUI interface may change by blocking access 
+    to certain GUI elements if the validation returns a false value.
+    """
+    def eventFilter(self, widget, event):
+        if event.type() == QtCore.QEvent.FocusIn:
+            widget.moveCursor(QtGui.QTextCursor.End, QTextCursor.MoveAnchor)
+            mcObj._d(6, "mc_d_label_3").setStyleSheet("background-color:#1c1c1c")
+            mcObj._d(1, "mc_d_paste_button").setEnabled(True)
+            mcObj.validated = False
+            return False
+        elif event.type() == QtCore.QEvent.FocusOut:
+            if mcObj._validate():
+                mcObj._d(6, "mc_d_label_3").setStyleSheet("background-color:darkgreen")
+            else:
+                mcObj._d(6, "mc_d_label_3").setStyleSheet("background-color:crimson")
+                mcObj._d(1, "mc_d_paste_button").setEnabled(False)
+            return False
+        else:
+            return False
+
+
+
+# Roman Numerals Conversion
+# Reference :  https://pypi.org/project/roman/
+# ------------------------------------------------------------------------------------------------
+
+romanNumeralMap = (
+    ("M", 1000),
+    ("CM", 900),
+    ("D", 500),
+    ("CD", 400),
+    ("C", 100),
+    ("XC", 90),
+    ("L", 50),
+    ("XL", 40),
+    ("X", 10),
+    ("IX", 9),
+    ("V", 5),
+    ("IV", 4),
+    ("I", 1),
+)
+
+def IntToRoman(n_):
+    result = ""
+    try:
+        for numeral, integer in romanNumeralMap:
+            while n_ >= integer:
+                result += numeral
+                n_ -= integer
+    except Exception:
+        result = ""
+    return result
+
+def RomanToInt(inputStr_):
+    result = 0
+    index = 0
+    try:
+        for numeral, integer in romanNumeralMap:
+            while inputStr_[index : index + len(numeral)] == numeral:
+                result += integer
+                index += len(numeral)
+    except Exception:
+        result = 0
+    return result
+
+
+
+# Alphabet Conversion
+# ------------------------------------------------------------------------------------------------
+
+alphaMap = [
+    RANDOMSTRING,
+    "a",
+    "b",
+    "c",
+    "d",
+    "e",
+    "f",
+    "g",
+    "h",
+    "i",
+    "j",
+    "k",
+    "l",
+    "m",
+    "n",
+    "o",
+    "p",
+    "q",
+    "r",
+    "s",
+    "t",
+    "u",
+    "v",
+    "w",
+    "x",
+    "y",
+    "z",
+]
+
+def IntToAlpha(n_):
+    result = ""
+    try:
+        if n_ >= 1 and n_ <= 26:
+            result = alphaMap[n_]
+        elif n_ >= 27 and n_ <= 702:
+            result = (
+                alphaMap[math.ceil((n_ - 26) / 26)]
+                + alphaMap[n_ - (26 * math.ceil((n_ - 26) / 26))]
+            )
+        elif n_ >= 703 and n_ <= 18278:
```

Comment:
What happens if n > 18278?

Reply:
I've thrown an exception. I don't think one can use more than 18278 in this macro.

Reply:
18278 also appears to be a high number but, you know, better safe than sorry. Where does this number come from?

Reply:
I agree, Mr. Galou.
So, 'a' corresponds to a '1', b to 2, z to 26, aa to 27, zz to 52, etc. 
Now, if the user enters a number, it will convert it to its alphabetic variant, or vice-versa.
But I don't think a user would have to perform a MultiCopy operation that'd go upto 18,278 copies; that would simply hang the application.